ckanext/datajson/blueprint.py

```diff
@@ -0,0 +1,416 @@
+                                      )
+    except p.toolkit.NotAuthorized:
+        logger.error('NotAuthorized to generate JSON for {} to {} ({})'.format(export_type, org_id, c.user))
+        auth = False
+
+    if not auth:
+        return "Not Authorized"
+
+    # set content type (charset required or pylons throws an error)
+    Response.content_type = 'application/json; charset=UTF-8'
+
+    # allow caching of response (e.g. by Apache)
+    # Commented because it works without it
+    # del Response.headers["Cache-Control"]
+    # del Response.headers["Pragma"]
+    resp = Response(make_json(export_type, org_id), mimetype='application/octet-stream')
+    resp.headers['Content-Disposition'] = 'attachment; filename="%s.zip"' % _zip_name
+
+    return resp
+
+
+def generate_output(fmt='json', org_id=None):
+    global _errors_json
+    _errors_json = []
+    # set content type (charset required or pylons throws an error)
+    Response.content_type = 'application/json; charset=UTF-8'
+
+    # allow caching of response (e.g. by Apache)
+    # Commented because it works without it
+    # del Response.headers["Cache-Control"]
+    # del Response.headers["Pragma"]
+
+    # TODO special processing for enterprise
+    # output
+    data = make_json(export_type='datajson', owner_org=org_id)
+
+    # if fmt == 'json-ld':
+    #     # Convert this to JSON-LD.
+    #     data = OrderedDict([
+    #         ("@context", OrderedDict([
+    #             ("rdfs", "http://www.w3.org/2000/01/rdf-schema#"),
+    #             ("dcterms", "http://purl.org/dc/terms/"),
+    #             ("dcat", "http://www.w3.org/ns/dcat#"),
+    #             ("foaf", "http://xmlns.com/foaf/0.1/"),
+    #         ])),
+    #         ("@id", DataJsonPlugin.ld_id),
+    #         ("@type", "dcat:Catalog"),
+    #         ("dcterms:title", DataJsonPlugin.ld_title),
+    #         ("rdfs:label", DataJsonPlugin.ld_title),
+    #         ("foaf:homepage", DataJsonPlugin.site_url),
+    #         ("dcat:dataset", [dataset_to_jsonld(d) for d in data.get('dataset')]),
+    #     ])
+
+    return p.toolkit.literal(json.dumps(data, indent=2))
+
+
+def make_json(export_type='datajson', owner_org=None):
+    # Error handler for creating error log
+    stream = io.StringIO()
+    eh = logging.StreamHandler(stream)
+    eh.setLevel(logging.WARN)
+    formatter = logging.Formatter('%(asctime)s - %(message)s')
+    eh.setFormatter(formatter)
+    logger.addHandler(eh)
+
+    data = ''
+    output = []
+    errors_json = []
+    Package2Pod.seen_identifiers = set()
+
+    try:
+        # Build the data.json file.
+        if owner_org:
+            if 'datajson' == export_type:
+                # we didn't check ownership for this type of export, so never load private datasets here
+                packages = _get_ckan_datasets(org=owner_org)
+                if not packages:
+                    packages = get_packages(owner_org=owner_org, with_private=False)
```

Comment:
This line should be commented out since we paginate the results in our fork. With this line the last pages will return all datasets.

We should also leave a comment explaining why it's commented out.